### PR TITLE
feat(Input.Password): password visibility controllable

### LINKED ISSUE #37999: 关于Input.Password组件密码明文和加密切换
### What problem does this feature solve?

例子：https://codepen.io/zhengjialux/pen/VwxgaqP
场景：在我们产品中有个专门修改密码的“页面”。修改密码后并不会去更新整个页面。
需求：
1、在密码框中输入密码后点击眼睛明文显示，input的type会变成text
2、点击保存后input的type并没有变回password模式，密码还是明文可见的。

### What does the proposed API look like?

希望加密和明文显示的状态开发者可控。

### PR DESCRIPTION
**Pull request by MrHeer**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #37999 

### 💡 Background and solution

update API
```ts
export type VisibilityToggle = {
  visible?: boolean;
  onVisibleChange?: (visible: boolean) => void;
};

export interface PasswordProps extends InputProps {
  ...
  visibilityToggle?: boolean | VisibilityToggle;
}
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Input.Password support controllable visible   |
| 🇨🇳 Chinese |     Input.Password 支持手动控制密码显隐       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/index.zh-CN.md [hunk 1]
```diff
@@ -87,6 +87,8 @@ Input 的其他属性和 React 自带的 [input](https://reactjs.org/docs/dom-el
 | --- | --- | --- | --- | --- |
 | iconRender | 自定义切换按钮 | (visible) => ReactNode | (visible) => (visible ? &lt;EyeOutlined /> : &lt;EyeInvisibleOutlined />) | 4.3.0 |
 | visibilityToggle | 是否显示切换按钮 | boolean | true |  |
+| visible | 用于手动控制密码显隐 | boolean | false | 4.24 |
+| onVisibleChange | 显隐密码的回调 | boolean | - | 4.24 |
```
Comment (afc163) [inline components/input/index.zh-CN.md:91]:
```suggestion
| passwordVisible | 用于手动控制密码显隐 | boolean | false | 4.24 |
| onPasswordVisibleChange | 显隐密码的回调 | boolean | - | 4.24 |
```

Comment (afc163) [reply components/input/index.zh-CN.md:91]: 想了下，这俩可以收在 visibilityToggle 里。

```
visibilityToggle={{ visible, onVisibleChange }}
```

Comment (MrHeer) [reply components/input/index.zh-CN.md:91]: 这个地方如果他是传入的对象是否需要显示 icon 呢？

Comment (afc163) [reply components/input/index.zh-CN.md:91]: 要

Comment (MrHeer) [reply components/input/index.zh-CN.md:91]: 可以增加这样一个类型吗？
```ts
export type VisibilityToggle =
  | boolean
  | {
      visible?: boolean;
      onVisibleChange?: (visible: boolean) => void;
    };
```

Comment (MrHeer) [reply components/input/index.zh-CN.md]: 我修改了，麻烦有空看一下

**Comment by afc163** [comment]
ci 挂了，合并一下 master

**Comment by MrHeer** [comment]
好像没办法直接合并，有 245 个文件冲突

**Comment by afc163** [comment]
合并一下基线分支，这里是 feature 分支

File: components/input/Password.tsx [hunk 2]
```diff
@@ -14,10 +14,15 @@ import Input from './Input';
 const defaultIconRender = (visible: boolean) =>
   visible ? <EyeOutlined /> : <EyeInvisibleOutlined />;
 
+export type VisibilityToggle = {
```
Comment (afc163) [inline components/input/Password.tsx:17]: 不需要 export

Comment (afc163) [inline components/input/Password.tsx:17]:
```suggestion
 type VisibilityToggle = {
```

**Review by afc163 — APPROVED** [review]

**Comment by PhilRandWu** [comment]
https://ant-design.antgroup.com/components/input-cn#visibilitytoggle

**Pull request merged**